### PR TITLE
HW 2 done

### PR DESCRIPTION
**Pull request by MrFiszbi1**
Here is my HW 2. Tell me what you think of my solution for question 3. I did it, but I don't know if that was the best way to go about it. For question 5 Any suggestions on how I can get the buttons to match the image exactly? I can't get it quite right.

### CODE REVIEW

File: hw2/03-count.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+const textHTML = text.innerHTML;
 
-input.addEventListener('keydown', handleKeyDown);
+input.addEventListener('input', handleKeyDown);
+
+function handleKeyDown(event) {
+  const value = event.target.value;
+  text.innerHTML = textHTML;
+  let textArray = textHTML.split(' ');
+  for (let count in textArray) {
```
Comment (capstone-psu-2023) [inline hw2/03-count.js]: another approach here involves using replaceAll(), might be worth looking at

File: hw2/04-background.js [hunk 2]
```diff
@@ -1 +1,50 @@
...
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
+function getRandomInt(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min) + min);
+}
+
+function changeColor() {
+  const background = document.querySelector('body');
+  const red = getRandomInt(0, 255);
```
Comment (capstone-psu-2023) [inline hw2/04-background.js]: not a big deal but I think your solution for getting a random int could be simpler, eg: 

const red = Math.floor(Math.random() * 255) 

this gives you an int between 0 and 255

File: hw2/04-background.js [hunk 3]
```diff
@@ -1 +1,50 @@
...
+  background.style.backgroundColor = `rgba(${red}, ${green}, ${blue}, 0.5)`;
+}
+
+function startIntervals(sec) {
+  const time = sec * 1000;
+  if (!nIntervId) {
+    nIntervId = setInterval(changeColor, time);
+  }
+  btn.textContent = 'Stop';
+  btn.style.backgroundColor = 'red';
```
Comment (capstone-psu-2023) [inline hw2/04-background.js]: maybe add bootstrap classes to the buttons rather than adjusting the style directly

File: hw2/05-form.html [hunk 4]
```diff
@@ -7,13 +7,116 @@
...
+        <div class="form-group row my-2">
+          <label for="textarea" class="col-12 col-form-label fw-bold"
+            >Anything else?</label
+          >
+          <div>
+            <textarea class="form-control" id="textarea" rows="3"></textarea>
+          </div>
+        </div>
+        <div class="text-center row d-flex justify-content-center">
+          <div class="col">
```
Comment (capstone-psu-2023) [inline hw2/05-form.html:115]: consider using d-flex class on the div encompassing the buttons to get  hem arranged like in the screenshot

File: hw2/04-background.html [hunk 5]
```diff
@@ -6,23 +6,26 @@
...
+      <p class="row mx-1">
         The background color will change every few seconds (as indicated below).
         You can stop this behavior and you can also change how often the
         background color changes.
       </p>
-      <div>
-        <label>Background change interval (seconds): </label>
-        <input />
+      <div class="row d-flex align-items-center">
+        <label class="row mx-1">Background change interval (seconds): </label>
```
Comment (capstone-psu-2023) [inline hw2/04-background.html]: you need a 'for' attribute on the label to connect it to the input to make this accessible

File: hw2/03-count.html [hunk 6]
```diff
@@ -6,19 +6,20 @@
...
   </head>
-  <body>
-    <main>
-      <h1>Exercise 03 - Count</h1>
-      <hr />
-      <div>
-        <label>Highlight all occurences of a word:</label>
-        <input />
+  <body class="bg-primary">
+    <main class="container vw-50 mt-4 bg-white rounded">
```
Comment (vpetrova123) [inline hw2/03-count.html]: You can do `<main class="container w-50 mx-auto mt-4 bg-white rounded">` here to style the box more like the HW screenshot.

File: hw2/03-count.js
Comment (vpetrova123) [inline hw2/03-count.js:1]: I agree with Matthew on this one. I used replaceAll() for this exercise. One more thing to note is that this solution doesn't work for words followed by a comma or a period. When I type "times" none of the instances get highlighted. I think using replaceAll() will help with this problem as well.

File: hw2/05-form.html
Comment (vpetrova123) [inline hw2/05-form.html:1]: To fix the buttons on this one you can do:
```
 <div class="form-group d-flex gap-1 my-3">
          <input type="submit" class="btn btn-primary w-50" value="Submit" />
          <input type="reset" class="btn btn-secondary w-50" value="Reset" />
 </div>
```

**Review by xuvea — COMMENTED** [review]
Everything is looking pretty good besides what capstone-psu-2023 has already pointed out. My only suggestion would be to look at using Bootstrap's card class instead of a container. It has a lot of predefined structure and styling to it that would accomplish a lot of the styling that was done in it's place.

File: hw2/05-form.js [hunk 7]
```diff
@@ -1 +1,66 @@
 // Add your code here
+let submission = document.querySelector('form');
+
+submission.addEventListener('submit', (event) => {
+  const title = '========= Form Submission =========';
+  const userName = document.getElementById('fullName');
+  const name = userName.value;
```
Comment (vpetrova123) [inline hw2/05-form.js]: You can simplify this by saying:
`const name = document.getElementById('fullName').value || 'no submission';`
That way if the value is not provided then the default value is 'no submission' and you don't have to do as many if/else statements later. The rest of the form inputs can be simplified in a similar way.

**Review by vpetrova123 — COMMENTED** [review]
Great job with the homework! I just had a few minor suggestions on bootstrap styling and simplifying some of the code.

**Pull request merged**